migration/src/main/java/kin/sdk/migration/MigrationManager.java

```diff
@@ -240,21 +280,47 @@ public void run() {
 	}
 
 	private void fireOnReady(final IMigrationManagerCallbacks migrationManagerCallbacks, final IKinClient kinClient,
-		final boolean needToSave) {
+		final String publicAddress, final boolean needToSave) {
 		handler.post(new Runnable() {
 			@Override
 			public void run() {
-				if (needToSave && kinClient.hasAccount()) {
-					IKinAccount account = kinClient.getAccount(kinClient.getAccountCount() - 1);
-					String publicAddress = account.getPublicAddress();
-					saveMigrationCompleted(publicAddress);
-				}
 				isMigrationInProcess.set(false);
-				if (migrationManagerCallbacks != null) {
-					migrationManagerCallbacks.onReady(kinClient);
+				boolean publicAddressIsEmpty = TextUtils.isEmpty(publicAddress);
+				// If no accounts and no public address then that mean we can supply a kinClient object to begin with.
+				// Or if the account has been found in the list of account then we can supply that kinClient which includes this account.
+				if (!kinClient.hasAccount() && publicAddressIsEmpty
```

Comment:
seems like a correct logic, but it's a bit confusing to me, I would expect to check first in the upper level and decide whether to call `fireOnReady` or `fireOnError` and not inside `fireOnReady` method.

Reply:
I thought that what you wanted because your comment was on the code in fireOnReady.
will check for a better place to do it

Reply:
anyway i just changed it so you can look at the commit.